### DISCUSSION TITLE
UK7004240 Radiator Covered reporting unsupported attribute & OTA fails

### DISCUSSION THREAD

**Discussion by UniqueName992** [Koenkk/zigbee2mqtt#11998] [Devices]
Hello!

On my [UK7004240](https://www.zigbee2mqtt.io/devices/UK7004240.html#hive-uk7004240) devices (Radiator valve based on Danfoss Ally (014G2461)) Z2M Exposes Radiator Covered and I'm trying to make use of it

I get this error back when I try and enable covered mode:
```
Publish 'set' 'radiator_covered' to 'PR Hive TRV' failed: 'Error: Write 0x14b457fffebed107/1 hvacThermostat({"danfossRadiatorCovered":true}, {"sendWhen":"fastpoll","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4678,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
```

I read that it's a recent firmware (see issue [10185](https://github.com/Koenkk/zigbee2mqtt/issues/10185)) for the 014G2461 devices that adds covered mode support, so I tried an OTA update, which returned me this error:
```
Failed to check if update available for 'PR Hive TRV' (No image available for imageType '66')
```
Does anyone have any hints on how to proceed, or is it simply not available for this device?

Coordinator: Conbee2
Revision: 0x26720700
FW Version: 0.6.71

Z2M Version: [1.24.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.24.0) commit: [a73caa3e](https://github.com/Koenkk/zigbee2mqtt/commit/a73caa3e)

TRV: [Hive](https://www.zigbee2mqtt.io/supported-devices/#v=Hive) [UK7004240](https://www.zigbee2mqtt.io/devices/UK7004240.html#hive-uk7004240)
Firmware build date: 20191127
Firmware version: 02.39.0008 02.39

**Comment by robertalexa** [comment]
Unfortunately not available on Hive. One thing (but i have no guarantee) is that Hive might have released a new version of their firmware. Which would mean pairing the TRV to hive, calling them and asking them to force update your devices, then pair them back to z2m and continue with your life without Hive's bullsh*t (I have 8 of these TRVs and i hate Hive, that is why i added all these features to every Hive thermostat related device and the Danfoss Ally)

I have a plan that i want to try, with the risk of completely bricking my devices. Force flashing the Hive TRV with the Danfoss Ally firmware which will then turn the device into a full on Danfoss Ally. However, no idea if this will work. I read the documentation about manual OTA updates, but the version numbers and device numbers are a bit confusing, maybe @Koenkk can give me some guidance. In all fairness i looked at this quite a few months ago, and have been waiting for winter to be over before i risk bricking my heating :D If this works, then i will make sure to include instructions in the documentation.

**Reply by wintersteiger** [reply]
I'm confused about the unsupported attributes on my Hive valves. @robertalexa did you go ahead with your plan of bricking you devices and what were your findings?

**Reply by fliespl** [reply]
@robertalexa did you ever try force updating hive to danfoss firmware?

**Reply by robertalexa** [reply]
I did and unfortunately not been successful. Here is the issue where i have worked on this with Koen
https://github.com/Koenkk/zigbee-OTA/issues/103
